src/Neo.IO/ByteArrayComparer.cs

```diff
@@ -20,27 +20,32 @@ internal class ByteArrayComparer : IComparer<byte[]>
         public static readonly ByteArrayComparer Default = new(1);
         public static readonly ByteArrayComparer Reverse = new(-1);
 
-        private readonly int direction;
+        private readonly int _direction;
 
         private ByteArrayComparer(int direction)
         {
-            this.direction = direction;
+            _direction = direction;
         }
 
-        public int Compare(byte[] x, byte[] y)
+        public int Compare(byte[]? x, byte[]? y)
         {
-            return direction > 0
-                ? CompareInternal(x, y)
-                : -CompareInternal(x, y);
+            if (ReferenceEquals(x, y)) return 0;
+            if (x is null && y is not null) return -y.Length;
+            if (y is null && x is not null) return x.Length;
+            if (x is not null && y is not null)
```

Comment:
Is imposible to be null x and y here, we can remove this if

Reply:
if they both null it return `0` its the last line in the method.

Reply:
But we check if both are null twice

```csharp
        public int Compare(byte[]? x, byte[]? y)
        {
            if (ReferenceEquals(x, y)) return 0; // x == y || x == null && y == null
            if (x is null && y is not null)
                return _direction > 0 ? -y.Length : y.Length;
            if (y is null && x is not null)
                return _direction > 0 ? x.Length : -x.Length;

#pragma warning disable CS8604 // Possible null reference argument.
            return _direction > 0 ?
                CompareInternal(x, y) :
                -CompareInternal(x, y);
#pragma warning restore CS8604 // Possible null reference argument.
        }
```